db/commitlog/commitlog.cc

```diff
@@ -1744,15 +1744,14 @@ future<> db::commitlog::segment_manager::delete_segments(std::vector<sstring> fi
     auto i = files.begin();
     auto e = files.end();
 
-    return parallel_for_each(i, e, [this](auto& filename) {
-        auto f = make_ready_future();
+    co_await parallel_for_each(i, e, [this](auto& filename) -> future<> {
         auto exts = cfg.extensions;
         if (exts && !exts->commitlog_file_extensions().empty()) {
-            f = parallel_for_each(exts->commitlog_file_extensions(), [&](auto& ext) {
-                return ext->before_delete(filename);
-            });
+            for (auto& ext : exts->commitlog_file_extensions()) {
+                co_await ext->before_delete(filename);
+            }
```

Comment:
This is not a pure translation, as the callbacks are now sequential. Though it's likely more correct (or maybe less wrong).

Reply:
The `before_delete` is only defined as exactly that, before delete (no ordering guarantee), but the above does not break it. And in fact should make any errors easier to pinpoint at least.